### DISCUSSION TITLE
Resolving CSS Aliases in Vite and Vanilla-Extract During Development

### DISCUSSION THREAD

**Discussion by f4-miyata** [vanilla-extract-css/vanilla-extract#1288] [Q&A]
Hello! Nice to meet you. I've recently started using Vite and vanilla-extract. When I set up aliases, I noticed that when running npm run build and previewing, the @assets in the url() within the css.ts file is resolved properly. However, during npm run dev, the aliases are not resolved, and url('@assets/...') remains as is. I've checked the issues, but I couldn't find a similar occurrence.

I created an example on CodeSandbox. No image is displayed behind the element that says "no logo !! in dev."

I'm not very familiar with English, so I apologize if my communication is unclear.
Thanks for your help! 

[codesandbox - vite](https://codesandbox.io/p/devbox/ex-forked-3smpjp)

![npm run dev](https://github.com/vanilla-extract-css/vanilla-extract/assets/45551590/b6b0af4f-3128-4976-af96-f568753a8fde)
![npm run build & preview](https://github.com/vanilla-extract-css/vanilla-extract/assets/45551590/c6905c17-6ecf-4792-9c91-fbecb385beb5)

By the way, aliases are resolved in development mode with Webpack as well.

[codesandbox - webpack](https://codesandbox.io/p/devbox/alias-in-webpack-vanilla-extract-gwdmkq)

**Comment by mrm007** [comment]
Copy/pasting the response from Discord:

Hi, this comes up every once in a while. We are aware it's not a great experience at the moment, and we have been working on a solution.

Previous discussion: https://github.com/vanilla-extract-css/vanilla-extract/discussions/1208
Community solution: https://github.com/vanilla-extract-css/vanilla-extract/issues/665#issuecomment-1342044702
This works with the current Vite plugin: https://github.com/wobsoriano/esbuild-vanilla-image-loader

The new Vite plugin will support this out of the box; no esbuild plugin needed. Because we aren't using esbuild any more and we're relying on the Vite CSS pipeline.
```ts
import myIcon from "../assets/icons/MyIcon.svg";

export const myListItem = style({
  listStyle: `url(${myIcon})`,
  // or
  listStyle: 'url("@assets/icons/MyIcon.svg")',
});
```
Please try it out and report any issues you find: https://github.com/vanilla-extract-css/vanilla-extract/pull/1264

I forked your sandbox with the new plugin: https://codesandbox.io/p/devbox/ex-forked-v7thjj
